### ISSUE TITLE
BUG: value counts with mixed NaNs is oddly rendering

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#45222]
in the 1.40rc1 release notes: https://pandas.pydata.org/pandas-docs/version/1.4/whatsnew/v1.4.0.html#null-values-are-no-longer-coerced-to-nan-value-in-value-counts-and-mode

We are seeing distinct `NaN` , but I think we should be seeing `NaN` and `NaT`

Looks like we patched here: https://github.com/pandas-dev/pandas/pull/42743

**Comment by jreback** [comment]
cc @realead if you can

**Comment by realead** [comment]
@jreback  Sorry I didn't pay the needed attention to the output and overlooked that when a series is printed all null-values are printed as `NaN`, i.e.:

```
import pandas as pd
import numpy as np
s=pd.Series([1, 2, 3, 4], [True, None, np.nan, pd.NaT])
```
and now 
 ```
print(s.keys())
print(s[np.nan])
 ```
work as expected:

```
Index([True, None, nan, NaT], dtype='object')
3
```
but 
```
print(s)
```
yields:
```
True    1
NaN     2
NaN     3
NaN     5
```

i.e. all `NaN`s.

For the documentation it is probably better to create another example. 

But is this  the behavior of series one would like to have/expect with `print`?

**Comment by realead** [comment]
I think this issue is closed via #45283
